File: Classic Algorithms/Collatz_Conjecture/collatz.py
"""
Collatz Conjecture:
    Start with a number n > 1. Find the number of steps it takes to
    reach one using the following process:
    If n is even, divide it by 2. If n is odd, multiply it by 3 and add 1
"""

try:
    n = int(input("Enter a number greater than one: "))
    if n <= 1:
        print("Invalid input")
        exit(0)
except ValueError:
    print("Invalid input")
    exit(0)

count = 0
while n != 1:
    print(n)
    if n % 2 == 0:
        n //= 2
        count += 1
    elif n % 2 > 0:
        n *= 3
        n += 1
        count += 1

print(n)
print("\nSteps taken: {}".format(count))

File: Classic Algorithms/Collatz_Conjecture/collatz.cpp
/*
Collatz Conjecture:
    Start with a number n > 1. Find the number of steps it takes to
    reach one using the following process:
    If n is even, divide it by 2. If n is odd, multiply it by 3 and add 1
*/

#include <iostream>
using std::cin; using std::cout; using std::endl;

int main() {
    int n, count = 0;
    cout << "Enter a number greater than one: ";
    cin >> n;

    while (n != 1) {
        cout << n << endl;
        if (n % 2 == 0) {
            n /= 2;
            ++count;
        }
        else if (n % 2 > 0) {
            n *= 3;
            ++n;
            ++count; 
        }
    }

    cout << n << endl;
    cout << "\nSteps taken: " << count << endl;

    return 0;
}